File: elements-of-programming-interviews/2023/cpp/reverse_digits.cc
#include "test_framework/generic_test.h"
long long Reverse(int x) {
  long long y = 0;
  long long neg = x < 0 ? -1 : 1;
  x *= neg;
  while (x) {
    y *= 10;
    y += x % 10;
    x /= 10;
  }
  return y * neg;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"x"};
  return GenericTestMain(args, "reverse_digits.cc", "reverse_digits.tsv",
                         &Reverse, DefaultComparator{}, param_names);
}

File: elements-of-programming-interviews/2023/python/reverse_digits.py
from test_framework import generic_test


def reverse(x: int) -> int:
    y = 0
    neg = -1 if x < 0 else 1
    x *= neg
    while x:
        y *= 10
        y += x % 10
        x = int(x/10)
    return y * neg


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('reverse_digits.py',
                                       'reverse_digits.tsv', reverse))
